src/linter/block_linter.go

```diff
@@ -569,6 +572,12 @@ func (b *blockLinter) checkContinueStmt(c *ir.ContinueStmt) {
 	}
 }
 
+func (b *blockLinter) checkArrayDimFetch(s *ir.ArrayDimFetchExpr) {
+	if s.CurlyBrace {
+		b.report(s, LevelDoNotReject, "arraySyntax", "Don't use curly braces to take an item by index (use square braces form instead)")
```

Comment:
Maybe this text would be easier to understand?
```
a{i} indexing is deprecated since PHP 7.4, use a[i] instead
```

Reply:
Done.